lib/internal/url.js

```diff
@@ -1406,14 +1408,26 @@ function getPathFromURL(path) {
 // as this is the only character that won't be percent encoded by
 // default URL percent encoding when pathname is set.
 const percentRegEx = /%/g;
-function getURLFromFilePath(filepath) {
+function pathToFileURL(filepath) {
+  let resolved = path.resolve(filepath);
+  // path.resolve strips trailing slashes so we must add them back
+  if (filepath[filepath.length - 1] === '/' &&
+      resolved[resolved.length - 1] !== '/')
+    resolved += '/';
   const tmp = new URL('file://');
```

Comment:
Does this handle Windows paths with `\\` separators?

Reply:
Yes, the WhatWG URL spec handles this for us.

Reply:
Do we have a test for the trailing slash behavior?